frontend/README.md

```diff
@@ -0,0 +1,389 @@
+
+## Design-System
+
+To ensure component reusability and avoid recreating the same components over and over, creating an set of initial `design-system` component which should provide the basic styles for commonly used controls like:
+
+1. Buttons
+2. Text
+3. Modals
+4. Colors
+
+### Style-Components
+
+The styled-component library becomes very handy when it comes to create extendible components based on a specific set of styles, it contains a `ThemeProvider` tha can be used to store the overall styling rules, like spacing, colors, fonts, etc.
+
+### Storybook
+
+The Storybook library is another great tool that can be used to create generic components while visualizing them before wiring them up with the rest of the application.
+
+## Application Entities
+
+- [Components](entities/Components.md)
+- [Models](entities/Models.md)
+- [Services](entities/Services.md)
+- [Gateways](entities/Gateways.md)
+- [Pages](entities/Pages.md)
+- [Types](entities/Types.md)
+
+## Rendering the Home Page
+
+Acceptance Criteria:
+
+AC1:
+As a user landing on the main page
+I should be able to see the loading state with skeleton items
+
+AC2:
+As a user landing on the main page
+And the requests are finish loading
+I should be able to see the image gallery
+
+AC3:
+As a user looking at the main page
+And I scroll to the bottom of the page
+I should be able to see the loading state when reaching the end
+And when the loading is done
+I should be able to see the second page of the images
+
+AC4:
+As a user looking at the main page
+And I scroll to the bottom of the page
+And I'm at the last page
+I should see the no more items component
+
+AC5:
+As a user looking at the main page
+I should be able to see the upload image button
+
+AC6:
+As a user looking at the main page
+And I click the upload button
+I should be prompted with the native controls to upload the image
+
+AC7:
+As a user looking at the main page
+And I select an image from the gallery or I took a photo
+I should be able to see the upload in progress
+And the website should be entirely functional
+
+AC8:
+As a user looking at the main page
+And I just uploaded an image
+Once the upload ends
+I should be able to see the updated list of images
+
+AC9:
+As a user looking at the main page
+And I just uploaded an image
+And I try to navigate away
+I should see an alert to confirm the navigation action
+
+### Component Tree
+
+Pseudo code of the main page component tree:
+
+```jsx
+<Layout>
+  <Navigation>
+  <ImageProvider>
+    <Page>
+      <Main>
+        <UploadInProgress>
+        <UploadButton>
+        <ImageList>
+          <ImageCard>
+          <NextPageTrigger>
+```
+
+### Pagination & Performance
+
+To avoid performance problems, the list of images will be paginated in chunks of 20 items, every time the user reaches the end, the skeleton/loading state should be displayed for the new section and a new set of images will be added when finish.
```

Comment:
```suggestion
To avoid performance problems the list of images will be paginated in chunks of 20 items. Every time the user reaches the end the skeleton/loading state should be displayed for the new section and a new set of images will be added when finished.
```
Will you ever "unload" items to conserve memory usage on the device?

Reply:
I think that if you are looking from a webapp perspective, not a native app, has a backend encoding mechanism for images and also has an async load for images and pagination, you can let the browser take care of it.